### ISSUE TITLE
The whole sentence disappears if you select a word.

### ISSUE THREAD

**Issue by tvvoty** [FooSoft/yomichan#1152]
Before if you passed a sentence into the search bar you could easily jump between words back and forth, but after the last update if you select a word the sentence disappears and you have to click "go back one page" in order to get it back. I guess that's a bug?

**Comment by toasted-nutbread** [comment]
When you say the sentence disappears, which sentence do you mean? The sentence in the textarea, or the parsed sentence below?

When you say "select a word", how are you selecting the word? E.g. holding modifier key + moving mouse over parsed sentence, or clicking on a word from the parsed sentence? (or other?)

From what I can tell from going back and forth a few versions, the behaviour is the same. Scanning a word from the sentence will replace the contents of the textarea with that word. This is confusing when using modifier+mouse as it doesn't create a new history entry, so I'll make a task to adjust that, but let me know if what I'm seeing is different from what you are seeing.

**Comment by tvvoty** [comment]
I was on windows, Firefox Version 84.0 (64-bit), just checked on lubuntu in VM and the bug isn't there. mb i'm missing something

> which sentence do you mean

I mean the parsed one below.
Before only the sentence in the
text area above would disappear but now they both do.

> how are you selecting the word

Clicking on a word from the parsed sentence with the left mouse button.
I actually didn’t know about the holding modifier key option
before now, I checked it and it works fine it seems, it doesn't replace the
sentence on the bottom but does in the text area, basically the way clicking worked before except the hovering.

> From what I can tell from going back and forth a few versions

As I said, checked it on Linux and it wasn't there, so I mb it's specific to my machine, or mb people don't use clicking that much.
Also, mb auto updates were off in my browser. For me it seemed the update was today\yesterday, both the new settings and
new interface. If that's not the case mb I was on much older version all this time and didn't realize.

> Scanning a word from the sentence will replace the contents of the textarea with that word. This is confusing when using modifier+mouse as it doesn't create a new history entry, so I'll make a task to adjust that, but let me know if what I'm seeing is different from what you are seeing

Yeah, it is kinda confusing and also you can't
recopy the sentence to paste it into other dictionary if you copied something else (which to me personally happened a lot since I copy screenshots after the sentence) so
that's prolly a good idea to change it, however, it always worked like that and it's not the problem i was talking about in the post.

basically, there’s summary:

before clicking on the word in a parsed sentence

![sentensebeforeclicking](https://user-images.githubusercontent.com/75815245/102743054-a42c6f00-4367-11eb-9eb8-bd19f7915749.PNG)


after clicking

![sentence after clicking](https://user-images.githubusercontent.com/75815245/102743107-d0e08680-4367-11eb-9c6e-b47977abbe07.PNG)


It's not that bad now that i know about hovering but still weird tho

**Comment by toasted-nutbread** [comment]
I'm having some difficulty reproducing the issue on my end is all, even on some previous releases. Could you export and share a settings file from the environment you are experiencing the issue in? Also, does this issue consistently happen every time you click on the parsed text, even after browser restart?

**Comment by toasted-nutbread** [comment]
Here's another idea, which if it is the problem, it is a bug with conflicting interactions:
On the settings page, try disabling this option (only need to do on one of the settings pages):
* On v1 page: `Enable search when clicking glossary entries and tags`
* On v2 page: `Search terms when clicking text from the results list`

**Comment by tvvoty** [comment]
I checked both setting pages and it was
enabled on v1 disabled on v2. 
I disabled it on v1 and it indeed solved the
issue! Thanks a lot!
Here's exported settings file before i changed them.
https://drive.google.com/file/d/1_Tp2MzYJaf3vqXeJ6SWj72knYrSngY9X/view?usp=sharing

> Also, does this issue consistently happen every time you click on the parsed text, even after browser restart?

Yeah, i tried restarting browser\pc and it would always behave that way.

**Comment by toasted-nutbread** [comment]
> I checked both setting pages and it was
> enabled on v1 disabled on v2.

Oops, that's another oversight that I'll fix also. Besides that, I've created a fix for the issue you had (conflicting settings) which will be included in the next release.

**Comment by tvvoty** [comment]
<3

**Comment by Keromaru5** [comment]
Hi, I've been experiencing this issue for a while, and the fixes in this thread and recent updates haven't resolved it. The sentence disappears on the search page both when I mouseover and when I click on a term.

**Comment by toasted-nutbread** [comment]
@Keromaru5 if you or anyone else is still able to consistently reproduce this issue, could you share an export of your settings file? I've seen this bug happen once, but I'm not sure how to reliably reproduce it for testing.

**Comment by Keromaru5** [comment]
Like this? [yomichan-settings-2021-06-26-04-45-28.txt](https://github.com/FooSoft/yomichan/files/6719581/yomichan-settings-2021-06-26-04-45-28.txt)

**Comment by toasted-nutbread** [comment]
Yes, thank you. Are you still consistently having the problem when you use the search page? Is there a specific way you are able to reproduce the issue / can you record a video of the issue happening?

**Comment by Keromaru5** [comment]
Yes, I still have the problem. Here's what it looks like.

https://user-images.githubusercontent.com/85845768/123518514-3e9da000-d674-11eb-96b5-51ed379f8b39.mp4

**Comment by toasted-nutbread** [comment]
The full sentence is still present in your video below the search box, is your comment in reference to the search box above it being "cleared" after selecting a term?

**Comment by Keromaru5** [comment]
That's right. It used to not do that. I'd prefer it to continue showing the whole sentence.

**Comment by toasted-nutbread** [comment]
The functionality has always been to replace the content of the search box with the selected term.
